shared/fs/container.js

```diff
@@ -58,12 +57,33 @@ const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps, ownPro
   }
 }
 
-export default compose(
+const ConnectedFiles = compose(
   connect(mapStateToProps, mapDispatchToProps, mergeProps),
+  setDisplayName('Files')
+)(Files)
+
+const FilesLoadingHoc = compose(
+  connect(
+    (state, {routeProps}) => ({
+      path: routeProps.get('path', Constants.defaultPath),
+    }),
+    (dispatch: Dispatch) => ({
+      loadFolderList: (path: Types.Path) => dispatch(FsGen.createFolderListLoad({path})),
+    }),
+    ({path}, {loadFolderList}) => ({
+      path,
+      loadFolderList,
+    })
+  ),
```

Comment:
Why can't these all be in the `Files` component above? The magic here is in the `componentWillUpdate` lifecycle below, right?

Reply:
I think if we do `componentWillUpdate` of the `Files` component, we’ll end up calling `loadFolderList` every time `Files` updates. Then the result of `loadFolderList` will cause state change which causes `Files` update again, which again triggers `loadFileList`. I guess it might converge eventually if we don’t update the immutable `children` when RPC result doesn’t change, but that’s still a bit wasteful.

With this, `loadFileList` only gets called when `routeProps` changes.

Reply:
There must be a cleaner way to do this...are these overrides canonical in our codebase?

Reply:
Not sure 🤔 I think I saw some HOCs in the code base but not sure about guidelines. I was hoping to converge to the right direction in code review.

Reply:
I made a few changes so that it doesn't use a full-blown `connect`. We only need `dispatch` in the HOC anyway. Hope it looks cleaner!